packages/core/src/cli/validate/upgradeability-assessment.ts

```diff
@@ -35,49 +18,34 @@ export interface UpgradeabilityAssessment {
 export function getUpgradeabilityAssessment(
   contract: SourceContract,
   allContracts: SourceContract[],
+  overrideReferenceContract?: SourceContract,
 ): UpgradeabilityAssessment {
   const fullContractName = contract.fullyQualifiedName;
   const contractValidation = contract.validationData[fullContractName];
 
   const isUUPS = inferUUPS(contract.validationData, fullContractName);
 
   const annotationAssessment = getAnnotationAssessment(contract);
-  if (annotationAssessment.upgradeable) {
-    let referenceContract = undefined;
-    let isReferenceUUPS = false;
-    if (annotationAssessment.referenceName !== undefined) {
-      referenceContract = getReferenceContract(annotationAssessment.referenceName, contract, allContracts);
-      isReferenceUUPS = inferUUPS(referenceContract.validationData, referenceContract.fullyQualifiedName);
-    }
 
-    return {
-      upgradeable: true,
-      referenceContract: referenceContract,
-      uups: isReferenceUUPS || isUUPS,
-    };
-  } else {
-    const initializable = inferInitializable(contractValidation);
-    return {
-      upgradeable: initializable || isUUPS,
-      uups: isUUPS,
-    };
+  let referenceContract = overrideReferenceContract ?? undefined;
```

Comment:
I don't see how this could be necessary?

```suggestion
  let referenceContract = overrideReferenceContract;
```